pandas/tests/series/test_arithmetic.py

```diff
@@ -679,6 +679,199 @@ def test_series_add_aware_naive_raises(self):
         with pytest.raises(Exception, match=msg):
             ser_utc + ser
 
+    @pytest.mark.filterwarnings(
+        "ignore:Adding/subtracting object-dtype array to DatetimeArray"
+    )
+    def test_series_add_daytime_offset(self):
```

Comment:
Rather than creating several tests, can you parametrize this over the different constructors within the same test to reduce code duplication? i.e., take the lists of timestamps and offsets and pass them to either Series, Index or DatetimeIndex, then do the arithmetic and assertions.

Since these would then be more generic arithmetic tests (not Series-specific) might want to put somewhere within pandas/tests/arithmetic. @jbrockmendel any preferences here?

Reply:
I have parameterized and moved the test. Hopefully I'm getting closer to a good solution.